src/levels.rs

```diff
@@ -316,27 +321,101 @@ pub struct Av1BlockIntra {
     pub cfl_alpha: [i8; 2],
 }
 
-#[derive(Clone, Copy)]
+/// Really an [`InterIntraPredMode`],
+/// but stored as a `u8` so there are no invalid bits.
+#[derive(Clone, Copy, PartialEq, Eq, FromZeroes, FromBytes, AsBytes)]
+#[repr(transparent)]
+pub struct MaskedInterIntraPredMode(u8);
+
+impl MaskedInterIntraPredMode {
+    pub const fn get(self) -> InterIntraPredMode {
+        // Should just be an `& 3`.
+        match InterIntraPredMode::from_repr(self.0 as usize % InterIntraPredMode::COUNT) {
```

Comment:
Why do we need to do `% InterIntraPredMode::COUNT`? Isn't `self.0` just the integer representation for `InterIntraPredMode`?

Reply:
The `u8`, since it's `FromBytes`, could be anything.  The `%` truncates off any invalid values so we only use the lower 2 bits that are the `InterIntraPredMode`.  Plus, even if the other bits are all `0` (if we use the `zerocopy` `union` correctly and don't actually reinterpret bytes), the compiler doesn't know this so it'll have to include a check and panic here, but by wrapping/truncating it in bounds, the compiler can elide the check.  Thus, it's just an `& 3`, which is essentially free.